pkgs/top-level/impure.nix

```diff
@@ -45,7 +45,7 @@ in
       overlays = dir:
         let content = readDir dir; in
         map (n: import (dir + ("/" + n)))
-          (builtins.filter (n: builtins.match ".*\.nix" n != null)
+          (builtins.filter (n: (builtins.match ".*\.nix" n != null) || pathExists (dir + ("/" + (n + "/default.nix"))))
```

Comment:
As mentioned in my comment:

> I'm using parenthesis to force the type to string because that seems to be the preferred style in this module. Also can't use ${} because of the old string ‘/nix/store/xxx/’ cannot refer to other paths

`dir` is of type path and `+` has a different meaning for path. I could also use `toString`